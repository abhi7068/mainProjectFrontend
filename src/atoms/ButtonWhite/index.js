import React from 'react';

import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
// re-defining material-ui classes for button
const useStyles = makeStyles((theme) => ({
  buttonWhite: {
    background: '#fff',
    boxShadow: '0 4px 6px 0 rgba(0,0,0,.12)',
    color: '#2874f0',
    fontWeight: '500',
    height: '48px',
  },
  contained: {
    '&:hover': {
      color: '#2874f0',
      boxShadow: '0 4px 6px 0 rgba(0,0,0,.12)',
      background: '#fff',
      cursor: 'pointer',
    },
  },
}));

function ButtonWhite({ children, ...rest }) {
  const classes = useStyles();
  return (
    <Button
      {...rest}
      className={classes.buttonWhite}
      classes={{ contained: classes.contained }}
    >
      {children}
    </Button>
  );
}
ButtonWhite.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
};
export default ButtonWhite;
