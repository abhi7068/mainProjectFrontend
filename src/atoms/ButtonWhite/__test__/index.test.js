import React from 'react';
import ButtonWhite from '../index';
import { render, cleanup } from '@testing-library/react';
afterEach(cleanup);
it('it renders', () => {
  render(<ButtonWhite />);
});
it('snapshot', () => {
  const { asFragment } = render(<ButtonWhite />);
  expect(asFragment()).toMatchSnapshot();
});
