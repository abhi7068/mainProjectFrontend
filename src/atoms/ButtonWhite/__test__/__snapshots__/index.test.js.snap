// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonWhite-45"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
